apps/src/applab/redux/data.js

```diff
@@ -20,14 +23,23 @@ const DataState = Record({
 const initialState = new DataState();
 
 export default function (state = initialState, action) {
+  let map;
   switch (action.type) {
+    case ADD_TABLE_NAME:
+      return state.set('tableListMap', Object.assign({}, state.tableListMap, {
+        [action.tableName]: true
+      }));
     case CHANGE_VIEW:
       // Discard table data when not viewing a table, so that we don't momentarily
       // show data for the wrong table when we return to the table view.
       if (action.view !== DataView.TABLE) {
         state = state.set('tableData', {});
       }
       return state.set('view', action.view).set('tableName', action.tableName);
+    case DELETE_TABLE_NAME:
+      map = Object.assign({}, state.tableListMap);
+      delete map[action.tableName];
+      return state.set('tableListMap', map);
```

Comment:
I wonder if we could/should do the same sort of thing here and just set map[action.tableName] to undefined. Up to you.
```
return state.set('tableListMap', Object.assign({}, state.tableListMap, {
  [action.tableName]: undefined
}));